### PR TITLE
fix: Fixes Horizontal Feature Loading States

### PR DESCRIPTION
**Pull request by redreceipt**
Fixes: https://public.3.basecamp.com/p/RJrwBjV4x1SnUypig3fMhQr6

Explicitly says that if title or subtitle is present in the horizontal list, hide loading state

### CODE REVIEW

File: packages/apollos-ui-kit/src/Placeholder/withPlaceholder.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ const withPlaceholder = (PlaceholderComponent = Line, getProps = {}) => (
       typeof getProps === 'function' ? getProps(props) : getProps;
     return (
       <PlaceholderComponent
-        onReady={!isLoading}
+        onReady={!isLoading || props.children}
```
Comment (redreceipt) [inline packages/apollos-ui-kit/src/Placeholder/withPlaceholder.js:15]: @vinnyjth I mention this in the description, this is optional here but I think could save us so much trouble down the road. And would probably allow us to take out A TON of edge case code for other components' loading states.

Comment (redreceipt) [reply packages/apollos-ui-kit/src/Placeholder/withPlaceholder.js:15]: it almost fixes this PRs problem completely except I think if we have subtitle, we don't need to show title loading, and vice versa, reason for the other file fix. but this isn't always the case for components loading states, most of the time, this placeholder fix would suffice

**Comment by redreceipt** [comment]
ok so all the snap changes are making me second guess the `withPlaceholder` change 😂 what am I missing here? how are children sneaking into the component?

File: packages/apollos-ui-kit/src/Placeholder/withPlaceholder.js (hunk 1 above)
Comment (vinnyjth) [reply packages/apollos-ui-kit/src/Placeholder/withPlaceholder.js:15]: Yah - this change is probably a little much. Our best bet is probably to always control the visibility of placeholders by testing for `isLoading` and finessing it further by changing the value of `isLoading`

**Comment by redreceipt** [comment]
ok @vinnyjth I put it all back :)

**Comment by vinnyjth** [comment]
Okay! So the only reason I _wouldn't_ want to merge this PR, is a UX issue. 

As a user, currently, when I pull to refresh, there are two indications that new content is loading in: 

1. I get placeholder loading states in my content items
2. My little pull to refresh indicator spins. 

The spinner logic is driven by the feed, so there's a period of time where a specific feature might still be reloading but the spinner is gone. For that reason, Think it might be valuable to continue to show placeholders _somewhere_ in the Feature if the feature is technically still loading. What do you think about keeping the logic above, by tweaking the logic of the rendered cards so that they are replaced with placeholder cards, while the feature is loading?

**Comment by redreceipt** [comment]
> What do you think about keeping the logic above, by tweaking the logic of the rendered cards so that they are replaced with placeholder cards, while the feature is loading?

I like that. So show cards loading but not titles.

**Comment by redreceipt** [comment]
wait haven't we seen weirdness with this before? I think it jumps in and out of loading states, something we've tried to get away from.

**Comment by vinnyjth** [comment]
@redreceipt We have, yah. That's likely due to any number of reasons, some of which we may have fixed and other we have not fixed. At one point it was due to how the loading state was passed down from the feed to each of the features, but I think that's a bit more stable now. Either way, we'll probably have to make sure it's something we can test for.

**Comment by redreceipt** [comment]
I’ll throw in the change and see if it’s a problem

**Comment by redreceipt** [comment]
so we could handle it two ways:

1. from this component, something like `content={isLoading ? null : cards}`

Screen Shot 2020-12-17 at 3 41 46 PM

or deeper in `TileFeed` itself but that will likely affect lots of components

**Comment by vinnyjth** [comment]
Handling it in this component sounds like the right approach. I'm not convinced this is the perfect final solution, but I think it's worth trying out and seeing how it feels :)

**Comment by redreceipt** [comment]
alright try it now

**Comment by vinnyjth** [comment]
@redreceipt Looking good - with one issue. 

What you'll see here is that when you pull to refetch - you can see "Sermons" temporarily. What's happening is that there's a sermon feed feature that's technically in the feed, but is hidden because it doesn't have any cards. 



In a production app - like newspring - that behavior won't check out fly. Some feeds (like the staff feeds) are in a similar state where they technically exist in the feed but most user's wont see them. 

I think the best case for this PR is to keep your `cards` loading logic (it looks really good!) and take out your tweaks to `title`/`subtitle` loading. If you have time, you could also apply that same `cards` logic to the other Features :)

**Comment by redreceipt** [comment]
try it now

**Comment by redreceipt** [comment]
Double-check you agree with this change on the Newspring app too. I don't really have an opinion either way.

**Review by vinnyjth — APPROVED** [review]

**Pull request merged**